### ISSUE TITLE
Fail to connect to sftp-server

### ISSUE THREAD

**Issue by assaft** [PowerShell/Win32-OpenSSH#1188]
**"OpenSSH for Windows" version**
7.6.0.1
 
**Server OperatingSystem**
Windows 10 Home

**Client OperatingSystem**
Windows 10 Pro

**What is failing**
I try to use WinSCP to connect to an OpenSSH for Windows. I tried both with user/pass and with a key. 
WinSCP fails with an error: 
> Cannot initialize SFTP protocol. Is the host running an SFTP server?

See attached sshd log. This line looks suspicious to me:
> 1252 2018-06-17 00:49:28.826 debug1: subsystem: cannot stat sftp-server.exe: No such file or directory

I have no problem to connect with PuTTY.

I have the following line in my sshd_config:
Subsystem	sftp	sftp-server.exe

**Actual output**

WinSCP fails with an error: 
> Cannot initialize SFTP protocol. Is the host running an SFTP server?

Server log:
[sshd.txt](https://github.com/PowerShell/Win32-OpenSSH/files/2108573/sshd.txt)

**Comment by assaft** [comment]
It could be that I simply encountered #1164 .

It seems like I don't manage to install version 7.7.1.0 and I'm not sure what I'm doing wrong.

I downloaded the latest version, which is supposed to be 7.7.1.0p1-Beta and install it as written below:

> PS C:\Program Files\OpenSSH> powershell.exe -ExecutionPolicy Bypass -File install-sshd.ps1
> [SC] SetServiceObjectSecurity SUCCESS
> [SC] ChangeServiceConfig2 SUCCESS
> [SC] ChangeServiceConfig2 SUCCESS
> sshd and ssh-agent services successfully installed
> PS C:\Program Files\OpenSSH> netsh advfirewall firewall add rule name=sshd dir=in action=allow protocol=TCP localport=22
> Ok.
> 
> PS C:\Program Files\OpenSSH> net start sshd
> The OpenSSH SSH Server service is starting.
> The OpenSSH SSH Server service was started successfully.
> 
> PS C:\Program Files\OpenSSH> Set-Service sshd -StartupType Automatic
> PS C:\Program Files\OpenSSH> Set-Service ssh-agent -StartupType Automatic
> PS C:\Program Files\OpenSSH> ((Get-Item (Get-Command sshd).Source).VersionInfo.FileVersion)
> 7.6.0.1
> PS C:\Program Files\OpenSSH>

As you can see, it prints 7.6.0.1.

It looks like I have two versions messed up - one in C:\Windows\System32\OpenSSH, which came with Windows built in OpenSSH, and the other in C:\Program Files\OpenSSH, which I installed manually. 

There seem to be two problems:
1. The built-in version hasn't been deleted when I uninstalled it from the menus.
2. The new version that I manually install doesn't take over the older version.

Could you please suggest how to work this out?

**Comment by assaft** [comment]
Regarding the version conflict:
1. I noticed that the PATH variable includes C:\Windows\System32\OpenSSH. This is a bug, since I already uninstalled the internal OpenSSH server.
2. After updating the PATH to C:\Program Files\OpenSSH and reinstalling, I get version 7.7.0.0 - still not  7.7.1.0. Could it be that the zip file for  7.7.1.0 Win64 is in fact 7.7.0.0?
3. WinSCP is still not working with 7.7.0.0.
4. I moved the directory to C:\OpenSSH and reinstalled. Now it works fine. This confirms that what I saw in the beginning was indeed #1164.

So to sum up:
1. There are problems related to uninstalling the internal OpenSSH server - the files are kept in System32\OpenSSH, and the PATH includes this directory. This causes a version conflict once the user installs other OpenSSH versions manually.

2. Although I try to use version 7.7.1.0 I get 7.7.0.0. 
This is the file I used: https://github.com/PowerShell/Win32-OpenSSH/releases/download/v7.7.1.0p1-Beta/OpenSSH-Win64.zip 

3. Moving the directory to C:\OpenSSH solved the original problem I had (#1164).

**Comment by manojampalam** [comment]
1. Did you uninstall OpenSSH client too? 

2. I downloaded your linked package, extracted and see the following:
```
PS C:\Temp\OpenSSH-Win64> (dir *.exe).VersionInfo
ProductVersion   FileVersion      FileName
OpenSSH_7.7p1... 7.7.1.0          C:\Temp\OpenSSH-Win64\scp.exe
OpenSSH_7.7p1... 7.7.1.0          C:\Temp\OpenSSH-Win64\sftp-server.exe
OpenSSH_7.7p1... 7.7.1.0          C:\Temp\OpenSSH-Win64\sftp.exe
OpenSSH_7.7p1... 7.7.1.0          C:\Temp\OpenSSH-Win64\ssh-add.exe
OpenSSH_7.7p1... 7.7.1.0          C:\Temp\OpenSSH-Win64\ssh-agent.exe
OpenSSH_7.7p1... 7.7.1.0          C:\Temp\OpenSSH-Win64\ssh-keygen.exe
OpenSSH_7.7p1... 7.7.1.0          C:\Temp\OpenSSH-Win64\ssh-keyscan.exe
OpenSSH_7.7p1... 7.7.1.0          C:\Temp\OpenSSH-Win64\ssh-shellhost.exe
OpenSSH_7.7p1... 7.7.1.0          C:\Temp\OpenSSH-Win64\ssh.exe
OpenSSH_7.7p1... 7.7.1.0          C:\Temp\OpenSSH-Win64\sshd.exe
```

**Comment by assaft** [comment]
1. No - I didn't uninstall the client.

2. It's really strange. It looks like I downloaded 7.7.1.0 but after running the installation script it somehow switched to 7.7.0.0.

This is what I get in the new path that I'm currently using (after running the installation script and everything as described in the wiki page):

```
PS C:\OpenSSH> dir


    Directory: C:\OpenSSH


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        6/17/2018   2:22 AM          15431 FixHostFilePermissions.ps1
-a----        6/17/2018   2:22 AM          13748 FixUserFilePermissions.ps1
-a----        6/17/2018   2:22 AM          15553 install-sshd.ps1
-a----        6/17/2018   2:22 AM        1624272 libcrypto.dll
-a----        6/17/2018   2:32 AM           2773 openssh-events.man
-a----        6/17/2018   2:22 AM          14199 OpenSSHUtils.psd1
-a----        6/17/2018   2:22 AM          40735 OpenSSHUtils.psm1
-a----        6/17/2018   2:22 AM         319176 scp.exe
-a----        6/17/2018   2:22 AM         331984 sftp-server.exe
-a----        6/17/2018   2:22 AM         396488 sftp.exe
-a----        6/17/2018   2:22 AM         497872 ssh-add.exe
-a----        6/17/2018   2:22 AM         393424 ssh-agent.exe
-a----        6/17/2018   2:22 AM         634064 ssh-keygen.exe
-a----        6/17/2018   2:22 AM         540368 ssh-keyscan.exe
-a----        6/17/2018   2:22 AM         158928 ssh-shellhost.exe
-a----        6/17/2018   2:22 AM         876240 ssh.exe
-a----        6/17/2018   2:22 AM         966856 sshd.exe
-a----        6/17/2018   2:22 AM           2143 sshd_config_default
-a----        6/17/2018   2:22 AM          13894 uninstall-sshd.ps1


PS C:\OpenSSH> (dir *.exe).VersionInfo

ProductVersion   FileVersion      FileName
--------------   -----------      --------
OpenSSH_7.7p1... 7.7.0.0          C:\OpenSSH\scp.exe
OpenSSH_7.7p1... 7.7.0.0          C:\OpenSSH\sftp-server.exe
OpenSSH_7.7p1... 7.7.0.0          C:\OpenSSH\sftp.exe
OpenSSH_7.7p1... 7.7.0.0          C:\OpenSSH\ssh-add.exe
OpenSSH_7.7p1... 7.7.0.0          C:\OpenSSH\ssh-agent.exe
OpenSSH_7.7p1... 7.7.0.0          C:\OpenSSH\ssh-keygen.exe
OpenSSH_7.7p1... 7.7.0.0          C:\OpenSSH\ssh-keyscan.exe
OpenSSH_7.7p1... 7.7.0.0          C:\OpenSSH\ssh-shellhost.exe
OpenSSH_7.7p1... 7.7.0.0          C:\OpenSSH\ssh.exe
OpenSSH_7.7p1... 7.7.0.0          C:\OpenSSH\sshd.exe
```

You can see that it comes from 7.7p1 but the version is 7.7.0.0. I'm not sure what's going on.

I tried to run these commands on the binaries right after extracting the ZIP file in my Downloads directory. This is what I got from that:

```
PS C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64> dir


    Directory: C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        6/17/2018   1:24 AM          15439 FixHostFilePermissions.ps1
-a----        6/17/2018   1:24 AM          13752 FixUserFilePermissions.ps1
-a----        6/17/2018   1:24 AM          15561 install-sshd.ps1
-a----        6/17/2018   1:24 AM        1624272 libcrypto.dll
-a----        6/17/2018   1:24 AM           6326 openssh-events.man
-a----        6/17/2018   1:24 AM          14231 OpenSSHUtils.psd1
-a----        6/17/2018   1:24 AM          40739 OpenSSHUtils.psm1
-a----        6/17/2018   1:24 AM         319184 scp.exe
-a----        6/17/2018   1:24 AM         331984 sftp-server.exe
-a----        6/17/2018   1:24 AM         395984 sftp.exe
-a----        6/17/2018   1:24 AM         497896 ssh-add.exe
-a----        6/17/2018   1:24 AM         393936 ssh-agent.exe
-a----        6/17/2018   1:24 AM         634064 ssh-keygen.exe
-a----        6/17/2018   1:24 AM         539856 ssh-keyscan.exe
-a----        6/17/2018   1:24 AM         158928 ssh-shellhost.exe
-a----        6/17/2018   1:24 AM         875728 ssh.exe
-a----        6/17/2018   1:24 AM         967888 sshd.exe
-a----        6/17/2018   1:24 AM           2143 sshd_config_default
-a----        6/17/2018   1:24 AM          13930 uninstall-sshd.ps1


PS C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64> (dir *.exe).VersionInfo

ProductVersion   FileVersion      FileName
--------------   -----------      --------
OpenSSH_7.7p1... 7.7.1.0          C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64\scp.exe
OpenSSH_7.7p1... 7.7.1.0          C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64\sftp-server.exe
OpenSSH_7.7p1... 7.7.1.0          C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64\sftp.exe
OpenSSH_7.7p1... 7.7.1.0          C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64\ssh-add.exe
OpenSSH_7.7p1... 7.7.1.0          C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64\ssh-agent.exe
OpenSSH_7.7p1... 7.7.1.0          C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64\ssh-keygen.exe
OpenSSH_7.7p1... 7.7.1.0          C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64\ssh-keyscan.exe
OpenSSH_7.7p1... 7.7.1.0          C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64\ssh-shellhost.exe
OpenSSH_7.7p1... 7.7.1.0          C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64\ssh.exe
OpenSSH_7.7p1... 7.7.1.0          C:\users\boriss\Downloads\OpenSSH-Win64 (1)\OpenSSH-Win64\sshd.exe
```

So here the files are indeed 7.7.1.0.

I don't know why it changes to 7.7.0.0 during the installation steps.

**Comment by manojampalam** [comment]
What are your installation steps? If you run install-sshd.ps1, it just installs the services and does nothing to binaries.

**Comment by assaft** [comment]
I started over again and now there's no problem with the version. It looks like there was some mistake on my side. Sorry about that.
